src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
+
+export const APIResponse = {
+  create: (
+    isSuccess: number,
+    statusCode: number,
+    message?: string,
+    data?: any,
+    error?: any,
+  ): ControllerResponse => {
+    const response: ControllerResponse = { isSuccess, statusCode, message };
+    if (error) {
+      response.error = error;
+    } else if (!error && data) {
+      response.data = data;
+    } else {
+      response.data = {};
+    }
+
+    return response;
+  },
```

Comment:
PullProofAI

The `APIResponse` object has a complex structure that can be simplified for better readability and maintainability.

```diff
- export const APIResponse = {
+ export const APIResponse = {
+   create: (isSuccess: number, statusCode: number, message?: string, data?: any, error?: any): ControllerResponse => {
+     const response: ControllerResponse = { isSuccess, statusCode, message };
+     if (error) {
+       response.error = error;
+     } else {
+       response.data = data || {};
+     }
+     return response;
+   },
+ };
```

Simplifying the logic within the `create` function by removing unnecessary conditions and improving variable names enhances code readability and reduces complexity.


<!-- This is an auto-generated comment by PullProofAI -->